synapse/events/utils.py

```diff
@@ -204,7 +203,9 @@ def _copy_field(src: JsonDict, dst: JsonDict, field: List[str]) -> None:
     key_to_move = field.pop(-1)
     sub_dict = src
     for sub_field in field:  # e.g. sub_field => "content"
-        if sub_field in sub_dict and type(sub_dict[sub_field]) in [dict, frozendict]:
```

Comment:
This one is slightly different, but I didn't see why it also shouldn't use `isinstance`?

Reply:
I guess this would reject `sub_dict[sub_fields]` which were instances of a class that inherits from either dict or frozendict. I don't think we have a particularly good reason to do that though?

Reply:
This should all be data from JSON, so I can't see how we would get anything besides `dict` or `frozendict`.

The code is from 2016 (70a2157b6458369b374cceeb0e5c8b0d985c6946 and 6d4e6d4cbac6f22457b2d5946c3dd7a7ea87ba3f) so I think it just didn't use `isinstance(...)`.